### PR TITLE
benchmark flamegraph save on s3

### PR DESCRIPTION
**Pull request by nkysg**
还需要把AWS_ACCESS_KEY通过secrets提供

### CODE REVIEW

**Review by jolestar — APPROVED** [review]

**Comment by jolestar** [comment]
https://github.com/starcoinorg/starcoin/runs/4310720756?check_suite_focus=true 
报错 

bash: ./scripts/flamegraph.sh: No such file or directory

感觉，是不是哪一步把 path 给改变了

**Comment by nkysg** [comment]
> https://github.com/starcoinorg/starcoin/runs/4310720756?check_suite_focus=true 报错
> 
> bash: ./scripts/flamegraph.sh: No such file or directory
> 
> 感觉，是不是哪一步把 pwd 给改变了

我想想看

**Comment by nkysg** [comment]
> > https://github.com/starcoinorg/starcoin/runs/4310720756?check_suite_focus=true 报错
> > bash: ./scripts/flamegraph.sh: No such file or directory
> > 感觉，是不是哪一步把 pwd 给改变了
> 
> 我想想看

换了顺序就好了，是不是重置下环境也行

File: .github/workflows/benchmark_pr.yml [hunk 1]
```diff
@@ -12,6 +12,15 @@ jobs:
...
       - name: setup rust toolchain
         uses: actions-rs/toolchain@v1
+      - name: run flamegraph
+        run: bash ./scripts/flamegraph.sh
+        env:
+          RUST_LOG: error
+          RUST_BACKTRACE: full
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - name: setup rust toolchain
```
Comment (jolestar) [inline .github/workflows/benchmark_pr.yml:22]: 前面已经 setup rust toolchain 了，没必要再 setup 一次。

**Comment by jolestar** [comment]
> > > https://github.com/starcoinorg/starcoin/runs/4310720756?check_suite_focus=true 报错
> > > bash: ./scripts/flamegraph.sh: No such file or directory
> > > 感觉，是不是哪一步把 pwd 给改变了
> > 
> > 
> > 我想想看
> 
> 换了顺序就好了，是不是重置下环境也行

可能是  boa-dev/criterion-compare-action@master 这步修改 pwd 了

File: .github/workflows/benchmark_pr.yml (hunk 1 above)
Comment (nkysg) [reply .github/workflows/benchmark_pr.yml]: 删除多余了

**Comment by nkysg** [comment]
pull request会运行benchmark_pr.yml 按pull request number作为文件夹生成火焰图
https://s3.ap-northeast-1.amazonaws.com/flamegraph.starcoin.org/3043/apply_block.svg
benchmark.yml按照commit_id为文件夹生成火焰图
https://s3.ap-northeast-1.amazonaws.com/flamegraph.starcoin.org/e78747e/apply_block.svg

**Pull request merged**